presto-main/src/test/java/com/facebook/presto/block/BlockAssertions.java

```diff
@@ -141,11 +141,13 @@ public static Block createStringsBlock(Iterable<String> values)
         return builder.build();
     }
 
-    public static Block createRandomStringBlock(int positionCount, boolean allowNulls, int maxStringLength)
+    public static Block createRandomStringBlock(int positionCount, float nullRate, int maxStringLength)
     {
+        verifyNullRate(nullRate);
+
         return createStringsBlock(
                 IntStream.range(0, positionCount)
-                        .mapToObj(i -> allowNulls && i % 7 == 1 ? null : generateRandomStringWithLength(maxStringLength))
+                        .mapToObj(i -> nullRate > 0 && ThreadLocalRandom.current().nextDouble(1) < nullRate ? null : generateRandomStringWithLength(maxStringLength))
```

Comment:
nit: consider extracting this logic into a helper class to avoid copy-paste

```
        ValuesWithNullsGenerator<String> generator = new ValuesWithNullsGenerator(nullRate, () -> generateRandomStringWithLength(maxStringLength));

        return createStringsBlock(
                IntStream.range(0, positionCount)
                        .mapToObj(i -> generator.next())
                        .collect(Collectors.toList()));

    private static final class ValuesWithNullsGenerator<T>
    {
        private final float nullRate;
        private final Supplier<T> supplier;

        private ValuesWithNullsGenerator(float nullRate, Supplier<T> supplier)
        {
            checkArgument (nullRate < 0 || nullRate > 1, "nullRate must be between 0 and 1 inclusive");

            this.nullRate = nullRate;
            this.supplier = requireNonNull(supplier, "supplier is null");
        }

        public T next()
        {
            return nullRate > 0 && ThreadLocalRandom.current().nextDouble(1) < nullRate ? null : supplier.get());
        }
    }
```